typescript/sdk/src/core/HyperlaneCoreChecker.ts

```diff
@@ -103,27 +103,50 @@ export class HyperlaneCoreChecker extends HyperlaneAppChecker<
       mailbox.address,
     );
 
-    await this.checkBytecode(
-      chain,
-      'Mailbox implementation',
-      implementation,
-      [
-        BytecodeHash.MAILBOX_WITHOUT_LOCAL_DOMAIN_BYTE_CODE_HASH,
-        BytecodeHash.MAILBOX_WITHOUT_LOCAL_DOMAIN_NONZERO_PAUSE_BYTE_CODE_HASH,
-      ],
-      (bytecode) =>
-        // This is obviously super janky but basically we are searching
-        //  for the ocurrences of localDomain in the bytecode and remove
-        //  that to compare, but some coincidental ocurrences of
-        // localDomain in the bytecode should be not be removed which
-        // are just done via an offset guard
-        bytecode.replaceAll(
-          ethersUtils.defaultAbiCoder
-            .encode(['uint32'], [localDomain])
-            .slice(2),
-          (match, offset) => (offset > 8000 ? match : ''),
-        ),
-    );
+    if (implementation === ethers.constants.AddressZero) {
+      const violation: MailboxViolation = {
+        type: CoreViolationType.Mailbox,
+        subType: MailboxViolationType.NotProxied,
+        contract: mailbox,
+        chain,
+        actual: implementation,
+        expected: 'non-zero address',
+      };
+      this.addViolation(violation);
+    } else {
+      await this.checkBytecode(
+        chain,
+        'Mailbox implementation',
+        implementation,
+        [
+          BytecodeHash.MAILBOX_WITHOUT_LOCAL_DOMAIN_BYTE_CODE_HASH,
+          BytecodeHash.MAILBOX_WITHOUT_LOCAL_DOMAIN_NONZERO_PAUSE_BYTE_CODE_HASH,
+          BytecodeHash.V3_MAILBOX_BYTECODE_HASH,
+        ],
+        (bytecode) =>
+          // This is obviously super janky but basically we are searching
+          //  for the ocurrences of localDomain in the bytecode and remove
+          //  that to compare, but some coincidental ocurrences of
+          // localDomain in the bytecode should be not be removed which
+          // are just done via an offset guard
```

Comment:
iiuc the immutables are always at the end of the bytecode (before the compiler metadata)
so maybe we should be a little more strict